flask_peewee/tests/test_rest.py

```diff
@@ -527,6 +545,102 @@ def test_delete(self):
         f_obj = FModel.get(id=self.f1.id)
         self.assertEqual(f_obj.e, None)
 
+    def test_json_filtering(self):
+        # test that filtering on JSON fields works
+        resp = self.app.get('/api/jmodel?j_field.bar=car')
+        self.assertAPIJModels(
+            resp.get_json(),
+            JModel.select().where(JModel.j_field['bar'] == 'car').order_by(JModel.id))
+
+    def test_json_value(self):
+        self.create_test_models()
+        # test wrong pk
+        resp = self.app.get('/api/jmodel/9000/j_field/new')
+        self.assertEqual(resp.status_code, 404)
+
+        # test wrong field name
+        resp = self.app.get('/api/jmodel/9000/no_field/new')
+        self.assertEqual(resp.status_code, 404)
+
+        url = '/api/jmodel/%s' % self.j1.id
+
+        resp = self.app.get(url + '/j_field/foo')
+        self.assertEqual(resp.status_code, 200)
+        assert resp.get_json() == {'star': 'far'}
+
+        resp = self.app.get(url + '/j_field/foo/star')
+        self.assertEqual(resp.status_code, 200)
+        assert resp.get_json() == 'far'
+
+        resp = self.app.get(url + '/j_field/foo/tzar')
+        self.assertEqual(resp.status_code, 200)
+        assert resp.get_json() == None
+
+        resp = self.app.get(url + '/j_field/foo/star/car')
+        self.assertEqual(resp.status_code, 200)
+        assert resp.get_json() == None
+
+    def test_json_edit(self):
+        self.create_test_models()
+        # test wrong pk
+        resp = self.app.put('/api/jmodel/9000/j_field/new', json='value')
+        self.assertEqual(resp.status_code, 404)
+
+        # test wrong field name
+        resp = self.app.put('/api/jmodel/9000/no_field/new', json='value')
+        self.assertEqual(resp.status_code, 404)
+
+        url = '/api/jmodel/%s' % self.j1.id
+        resp = self.app.put(url + '/j_field/new', json='value')
+        self.assertEqual(resp.status_code, 200)
+        j1 = JModel.get(JModel.id == self.j1.id)
+        assert resp.get_json() == 'value'
+        assert j1.j_field == {'foo': {'star': 'far'}, 'bar': 'car', 'new': 'value'}
+
+        resp = self.app.put(url + '/j_field/foo', json={'star': 'sun'})
```

Comment:
How about changing to `json={'planet': 'mars'}` to demonstrate whether this is an overwrite or merge with existing value?